dm/master/scheduler/scheduler.go

```diff
@@ -1198,14 +1215,31 @@ func (s *Scheduler) tryBoundForWorker(w *Worker) (bounded bool, err error) {
 // tryBoundForSource tries to bound a source to a random Free worker.
 // returns (true, nil) after bounded.
 func (s *Scheduler) tryBoundForSource(source string) (bool, error) {
-	// 1. try to find a random Free worker.
+	// 1. try to find history workers, then random Free worker.
```

Comment:
Seems no need now. Because we now remove lastbound when stop-source. And if a worker offline and call `tryBoundForSource`, this check make no sense

Reply:
if a worker offline and call `tryBoundForSource`, history worker is still prefered when

(s1, w1), (null, w2), (null, w3)
worker1 offline
(s1, w2), (null, w3)
worker1 online
(null, w1), (s1, w2), (null, w3)

now user could kill w2 to wish w1 bounded to s1. and the priority could ensure that scheduling.

Reply:
> (s1, w2), (null, w3)

Will lastbound be overwritten here?

Reply:
the key of lastbound is worker, so not overwritten